### ISSUE TITLE
sun4i build of tools/ fails due to including /usr/include/image.h instead of local image.h

### ISSUE THREAD

**Issue by codyps** [linux-sunxi/u-boot-sunxi#50]
Essentially, the include dependencies are a mess. tools/Makefile uses -idirafter for several include directories in the hope of getting the libfdt_env.h include to work correctly, but this causes the /usr/include/image.h to take precedence over $(SRCTREE)/include/image.h , which breaks the build.

Full build log follows:

```
nyx@zero /home/nyx/sunxi/u-boot % make CROSS_COMPILE=arm-linux-gnueabihf- sun4i
Configuring for sun4i - Board: sun4i, Options: SUNXI_EMAC
make
make[1]: Entering directory `/home/nyx/sunxi/u-boot'
Generating include/autoconf.mk
Generating include/autoconf.mk.dep
make[1]: Leaving directory `/home/nyx/sunxi/u-boot'
make[1]: Entering directory `/home/nyx/sunxi/u-boot'
Generating include/spl-autoconf.mk
Generating include/tpl-autoconf.mk
arm-linux-gnueabihf-gcc -DDO_DEPS_ONLY \
    -g  -Os   -ffunction-sections -fdata-sections -fno-common -ffixed-r8 -msoft-float  -D__KERNEL__ -DCONFIG_SYS_TEXT_BASE=0x4a000000 -I/home/nyx/sunxi/u-boot/include -fno-builtin -ffreestanding -nostdinc -isystem /usr/lib/gcc/arm-linux-gnueabihf/4.7.3/include -pipe  -DCONFIG_ARM -D__ARM__ -marm -mno-thumb-interwork -mabi=aapcs-linux -march=armv7-a -Wall -Wstrict-prototypes -fno-stack-protector -Wno-format-nonliteral -Wno-format-security -fstack-usage   \
    -o lib/asm-offsets.s lib/asm-offsets.c -c -S
Generating include/generated/generic-asm-offsets.h
tools/scripts/make-asm-offsets lib/asm-offsets.s include/generated/generic-asm-offsets.h
if [ -f arch/arm/cpu/armv7/sunxi/asm-offsets.c ];then \
    arm-linux-gnueabihf-gcc -DDO_DEPS_ONLY \
    -g  -Os   -ffunction-sections -fdata-sections -fno-common -ffixed-r8 -msoft-float  -D__KERNEL__ -DCONFIG_SYS_TEXT_BASE=0x4a000000 -I/home/nyx/sunxi/u-boot/include -fno-builtin -ffreestanding -nostdinc -isystem /usr/lib/gcc/arm-linux-gnueabihf/4.7.3/include -pipe  -DCONFIG_ARM -D__ARM__ -marm -mno-thumb-interwork -mabi=aapcs-linux -march=armv7-a -Wall -Wstrict-prototypes -fno-stack-protector -Wno-format-nonliteral -Wno-format-security -fstack-usage   \
        -o arch/arm/cpu/armv7/sunxi/asm-offsets.s arch/arm/cpu/armv7/sunxi/asm-offsets.c -c -S; \
else \
    touch arch/arm/cpu/armv7/sunxi/asm-offsets.s; \
fi
Generating include/generated/asm-offsets.h
tools/scripts/make-asm-offsets arch/arm/cpu/armv7/sunxi/asm-offsets.s include/generated/asm-offsets.h
for dir in tools examples/standalone examples/api arch/arm/cpu/armv7  ; do \
        make -C $dir _depend ; done
make[2]: Entering directory `/home/nyx/sunxi/u-boot/tools'
cat /dev/null  >.depend
make[2]: Leaving directory `/home/nyx/sunxi/u-boot/tools'
make[2]: Entering directory `/home/nyx/sunxi/u-boot/tools'
make[2]: Nothing to be done for `_depend'.
make[2]: Leaving directory `/home/nyx/sunxi/u-boot/tools'
make[2]: Entering directory `/home/nyx/sunxi/u-boot/examples/standalone'
arm-linux-gnueabihf-gcc -M -g  -Os   -ffunction-sections -fdata-sections -fno-common -ffixed-r8 -msoft-float  -D__KERNEL__ -DCONFIG_SYS_TEXT_BASE=0x4a000000 -I/home/nyx/sunxi/u-boot/include -fno-builtin -ffreestanding -nostdinc -isystem /usr/lib/gcc/arm-linux-gnueabihf/4.7.3/include -pipe  -DCONFIG_ARM -D__ARM__ -marm -mno-thumb-interwork -mabi=aapcs-linux -march=armv7-a -I..   -MQ hello_world.o hello_world.c >.depend.hello_world
arm-linux-gnueabihf-gcc -M -g  -Os   -ffunction-sections -fdata-sections -fno-common -ffixed-r8 -msoft-float  -D__KERNEL__ -DCONFIG_SYS_TEXT_BASE=0x4a000000 -I/home/nyx/sunxi/u-boot/include -fno-builtin -ffreestanding -nostdinc -isystem /usr/lib/gcc/arm-linux-gnueabihf/4.7.3/include -pipe  -DCONFIG_ARM -D__ARM__ -marm -mno-thumb-interwork -mabi=aapcs-linux -march=armv7-a -I..   -MQ stubs.o stubs.c >.depend.stubs
cat /dev/null .depend.hello_world .depend.stubs >.depend
make[2]: Leaving directory `/home/nyx/sunxi/u-boot/examples/standalone'
make[2]: Entering directory `/home/nyx/sunxi/u-boot/examples/standalone'
make[2]: Nothing to be done for `_depend'.
make[2]: Leaving directory `/home/nyx/sunxi/u-boot/examples/standalone'
make[2]: Entering directory `/home/nyx/sunxi/u-boot/examples/api'
cat /dev/null  >.depend
make[2]: Leaving directory `/home/nyx/sunxi/u-boot/examples/api'
make[2]: Entering directory `/home/nyx/sunxi/u-boot/examples/api'
make[2]: Nothing to be done for `_depend'.
make[2]: Leaving directory `/home/nyx/sunxi/u-boot/examples/api'
make[2]: Entering directory `/home/nyx/sunxi/u-boot/arch/arm/cpu/armv7'
arm-linux-gnueabihf-gcc -M -g  -Os   -ffunction-sections -fdata-sections -fno-common -ffixed-r8 -msoft-float  -D__KERNEL__ -DCONFIG_SYS_TEXT_BASE=0x4a000000 -I/home/nyx/sunxi/u-boot/include -fno-builtin -ffreestanding -nostdinc -isystem /usr/lib/gcc/arm-linux-gnueabihf/4.7.3/include -pipe  -DCONFIG_ARM -D__ARM__ -marm -mno-thumb-interwork -mabi=aapcs-linux -march=armv7-a   -MQ start.o start.S >.depend.start
arm-linux-gnueabihf-gcc -M -g  -Os   -ffunction-sections -fdata-sections -fno-common -ffixed-r8 -msoft-float  -D__KERNEL__ -DCONFIG_SYS_TEXT_BASE=0x4a000000 -I/home/nyx/sunxi/u-boot/include -fno-builtin -ffreestanding -nostdinc -isystem /usr/lib/gcc/arm-linux-gnueabihf/4.7.3/include -pipe  -DCONFIG_ARM -D__ARM__ -marm -mno-thumb-interwork -mabi=aapcs-linux -march=armv7-a   -MQ cache_v7.o cache_v7.c >.depend.cache_v7
arm-linux-gnueabihf-gcc -M -g  -Os   -ffunction-sections -fdata-sections -fno-common -ffixed-r8 -msoft-float  -D__KERNEL__ -DCONFIG_SYS_TEXT_BASE=0x4a000000 -I/home/nyx/sunxi/u-boot/include -fno-builtin -ffreestanding -nostdinc -isystem /usr/lib/gcc/arm-linux-gnueabihf/4.7.3/include -pipe  -DCONFIG_ARM -D__ARM__ -marm -mno-thumb-interwork -mabi=aapcs-linux -march=armv7-a   -MQ cpu.o cpu.c >.depend.cpu
arm-linux-gnueabihf-gcc -M -g  -Os   -ffunction-sections -fdata-sections -fno-common -ffixed-r8 -msoft-float  -D__KERNEL__ -DCONFIG_SYS_TEXT_BASE=0x4a000000 -I/home/nyx/sunxi/u-boot/include -fno-builtin -ffreestanding -nostdinc -isystem /usr/lib/gcc/arm-linux-gnueabihf/4.7.3/include -pipe  -DCONFIG_ARM -D__ARM__ -marm -mno-thumb-interwork -mabi=aapcs-linux -march=armv7-a   -MQ syslib.o syslib.c >.depend.syslib
arm-linux-gnueabihf-gcc -M -g  -Os   -ffunction-sections -fdata-sections -fno-common -ffixed-r8 -msoft-float  -D__KERNEL__ -DCONFIG_SYS_TEXT_BASE=0x4a000000 -I/home/nyx/sunxi/u-boot/include -fno-builtin -ffreestanding -nostdinc -isystem /usr/lib/gcc/arm-linux-gnueabihf/4.7.3/include -pipe  -DCONFIG_ARM -D__ARM__ -marm -mno-thumb-interwork -mabi=aapcs-linux -march=armv7-a   -MQ cmd_boot.o cmd_boot.c >.depend.cmd_boot
cat /dev/null .depend.start .depend.cache_v7 .depend.cpu .depend.syslib .depend.cmd_boot >.depend
make[2]: Leaving directory `/home/nyx/sunxi/u-boot/arch/arm/cpu/armv7'
make[2]: Entering directory `/home/nyx/sunxi/u-boot/arch/arm/cpu/armv7'
make[2]: Nothing to be done for `_depend'.
make[2]: Leaving directory `/home/nyx/sunxi/u-boot/arch/arm/cpu/armv7'
make -C tools all
make[2]: Entering directory `/home/nyx/sunxi/u-boot/tools'
gcc -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer -include /home/nyx/sunxi/u-boot/include/libfdt_env.h -idirafter /home/nyx/sunxi/u-boot/include -idirafter /home/nyx/sunxi/u-boot/include2 -idirafter /home/nyx/sunxi/u-boot/include -I /home/nyx/sunxi/u-boot/lib/libfdt -I /home/nyx/sunxi/u-boot/tools -DCONFIG_SYS_TEXT_BASE=0x4a000000 -DUSE_HOSTCC -D__KERNEL_STRICT_NAMES -D_GNU_SOURCE -pedantic   -o gen_eth_addr.o gen_eth_addr.c -c
gcc -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer -include /home/nyx/sunxi/u-boot/include/libfdt_env.h -idirafter /home/nyx/sunxi/u-boot/include -idirafter /home/nyx/sunxi/u-boot/include2 -idirafter /home/nyx/sunxi/u-boot/include -I /home/nyx/sunxi/u-boot/lib/libfdt -I /home/nyx/sunxi/u-boot/tools -DCONFIG_SYS_TEXT_BASE=0x4a000000 -DUSE_HOSTCC -D__KERNEL_STRICT_NAMES -D_GNU_SOURCE -pedantic  -o gen_eth_addr gen_eth_addr.o
strip gen_eth_addr
gcc -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer -include /home/nyx/sunxi/u-boot/include/libfdt_env.h -idirafter /home/nyx/sunxi/u-boot/include -idirafter /home/nyx/sunxi/u-boot/include2 -idirafter /home/nyx/sunxi/u-boot/include -I /home/nyx/sunxi/u-boot/lib/libfdt -I /home/nyx/sunxi/u-boot/tools -DCONFIG_SYS_TEXT_BASE=0x4a000000 -DUSE_HOSTCC -D__KERNEL_STRICT_NAMES -D_GNU_SOURCE -pedantic   -o img2srec.o img2srec.c -c
gcc -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer -include /home/nyx/sunxi/u-boot/include/libfdt_env.h -idirafter /home/nyx/sunxi/u-boot/include -idirafter /home/nyx/sunxi/u-boot/include2 -idirafter /home/nyx/sunxi/u-boot/include -I /home/nyx/sunxi/u-boot/lib/libfdt -I /home/nyx/sunxi/u-boot/tools -DCONFIG_SYS_TEXT_BASE=0x4a000000 -DUSE_HOSTCC -D__KERNEL_STRICT_NAMES -D_GNU_SOURCE -pedantic  -o img2srec img2srec.o
strip img2srec
gcc -g -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer -include /home/nyx/sunxi/u-boot/include/libfdt_env.h -idirafter /home/nyx/sunxi/u-boot/include -idirafter /home/nyx/sunxi/u-boot/include2 -idirafter /home/nyx/sunxi/u-boot/include -I /home/nyx/sunxi/u-boot/lib/libfdt -I /home/nyx/sunxi/u-boot/tools -DCONFIG_SYS_TEXT_BASE=0x4a000000 -DUSE_HOSTCC -D__KERNEL_STRICT_NAMES -D_GNU_SOURCE -pedantic -c -o crc32.o /home/nyx/sunxi/u-boot/lib/crc32.c
gcc -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer -include /home/nyx/sunxi/u-boot/include/libfdt_env.h -idirafter /home/nyx/sunxi/u-boot/include -idirafter /home/nyx/sunxi/u-boot/include2 -idirafter /home/nyx/sunxi/u-boot/include -I /home/nyx/sunxi/u-boot/lib/libfdt -I /home/nyx/sunxi/u-boot/tools -DCONFIG_SYS_TEXT_BASE=0x4a000000 -DUSE_HOSTCC -D__KERNEL_STRICT_NAMES -D_GNU_SOURCE   -o mkenvimage.o mkenvimage.c -c
gcc -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer -include /home/nyx/sunxi/u-boot/include/libfdt_env.h -idirafter /home/nyx/sunxi/u-boot/include -idirafter /home/nyx/sunxi/u-boot/include2 -idirafter /home/nyx/sunxi/u-boot/include -I /home/nyx/sunxi/u-boot/lib/libfdt -I /home/nyx/sunxi/u-boot/tools -DCONFIG_SYS_TEXT_BASE=0x4a000000 -DUSE_HOSTCC -D__KERNEL_STRICT_NAMES -D_GNU_SOURCE   -o os_support.o os_support.c -c
gcc -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer -include /home/nyx/sunxi/u-boot/include/libfdt_env.h -idirafter /home/nyx/sunxi/u-boot/include -idirafter /home/nyx/sunxi/u-boot/include2 -idirafter /home/nyx/sunxi/u-boot/include -I /home/nyx/sunxi/u-boot/lib/libfdt -I /home/nyx/sunxi/u-boot/tools -DCONFIG_SYS_TEXT_BASE=0x4a000000 -DUSE_HOSTCC -D__KERNEL_STRICT_NAMES -D_GNU_SOURCE -pedantic  -o mkenvimage crc32.o mkenvimage.o os_support.o
strip mkenvimage
gcc -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer -include /home/nyx/sunxi/u-boot/include/libfdt_env.h -idirafter /home/nyx/sunxi/u-boot/include -idirafter /home/nyx/sunxi/u-boot/include2 -idirafter /home/nyx/sunxi/u-boot/include -I /home/nyx/sunxi/u-boot/lib/libfdt -I /home/nyx/sunxi/u-boot/tools -DCONFIG_SYS_TEXT_BASE=0x4a000000 -DUSE_HOSTCC -D__KERNEL_STRICT_NAMES -D_GNU_SOURCE   -o aisimage.o aisimage.c -c
In file included from /usr/include/image.h:22:0,
                 from aisimage.c:10:
/usr/include/rgbpixel.h:61:3: error: expected specifier-qualifier-list before ‘RGBpixel’
In file included from aisimage.c:10:0:
/usr/include/image.h:40:1: error: unknown type name ‘class’
/usr/include/image.h:41:1: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘{’ token
aisimage.c:22:1: error: unknown type name ‘table_entry_t’
aisimage.c:23:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:23:2: warning: (near initialization for ‘aisimage_cmds[0]’) [enabled by default]
aisimage.c:23:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:23:2: warning: (near initialization for ‘aisimage_cmds[0]’) [enabled by default]
aisimage.c:23:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:23:2: warning: (near initialization for ‘aisimage_cmds[0]’) [enabled by default]
aisimage.c:24:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:24:2: warning: (near initialization for ‘aisimage_cmds[1]’) [enabled by default]
aisimage.c:24:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:24:2: warning: (near initialization for ‘aisimage_cmds[1]’) [enabled by default]
aisimage.c:24:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:24:2: warning: (near initialization for ‘aisimage_cmds[1]’) [enabled by default]
aisimage.c:25:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:25:2: warning: (near initialization for ‘aisimage_cmds[2]’) [enabled by default]
aisimage.c:25:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:25:2: warning: (near initialization for ‘aisimage_cmds[2]’) [enabled by default]
aisimage.c:25:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:25:2: warning: (near initialization for ‘aisimage_cmds[2]’) [enabled by default]
aisimage.c:26:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:26:2: warning: (near initialization for ‘aisimage_cmds[3]’) [enabled by default]
aisimage.c:26:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:26:2: warning: (near initialization for ‘aisimage_cmds[3]’) [enabled by default]
aisimage.c:26:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:26:2: warning: (near initialization for ‘aisimage_cmds[3]’) [enabled by default]
aisimage.c:27:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:27:2: warning: (near initialization for ‘aisimage_cmds[4]’) [enabled by default]
aisimage.c:27:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:27:2: warning: (near initialization for ‘aisimage_cmds[4]’) [enabled by default]
aisimage.c:27:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:27:2: warning: (near initialization for ‘aisimage_cmds[4]’) [enabled by default]
aisimage.c:28:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:28:2: warning: (near initialization for ‘aisimage_cmds[5]’) [enabled by default]
aisimage.c:28:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:28:2: warning: (near initialization for ‘aisimage_cmds[5]’) [enabled by default]
aisimage.c:28:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:28:2: warning: (near initialization for ‘aisimage_cmds[5]’) [enabled by default]
aisimage.c:29:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:29:2: warning: (near initialization for ‘aisimage_cmds[6]’) [enabled by default]
aisimage.c:29:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:29:2: warning: (near initialization for ‘aisimage_cmds[6]’) [enabled by default]
aisimage.c:29:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:29:2: warning: (near initialization for ‘aisimage_cmds[6]’) [enabled by default]
aisimage.c:30:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:30:2: warning: (near initialization for ‘aisimage_cmds[7]’) [enabled by default]
aisimage.c:30:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:30:2: warning: (near initialization for ‘aisimage_cmds[7]’) [enabled by default]
aisimage.c:30:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:30:2: warning: (near initialization for ‘aisimage_cmds[7]’) [enabled by default]
aisimage.c:31:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:31:2: warning: (near initialization for ‘aisimage_cmds[8]’) [enabled by default]
aisimage.c:31:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:31:2: warning: (near initialization for ‘aisimage_cmds[8]’) [enabled by default]
aisimage.c:31:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:31:2: warning: (near initialization for ‘aisimage_cmds[8]’) [enabled by default]
aisimage.c:32:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:32:2: warning: (near initialization for ‘aisimage_cmds[9]’) [enabled by default]
aisimage.c:32:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:32:2: warning: (near initialization for ‘aisimage_cmds[9]’) [enabled by default]
aisimage.c:32:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:32:2: warning: (near initialization for ‘aisimage_cmds[9]’) [enabled by default]
aisimage.c:33:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:33:2: warning: (near initialization for ‘aisimage_cmds[10]’) [enabled by default]
aisimage.c:33:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:33:2: warning: (near initialization for ‘aisimage_cmds[10]’) [enabled by default]
aisimage.c:33:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:33:2: warning: (near initialization for ‘aisimage_cmds[10]’) [enabled by default]
aisimage.c:34:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:34:2: warning: (near initialization for ‘aisimage_cmds[11]’) [enabled by default]
aisimage.c:34:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:34:2: warning: (near initialization for ‘aisimage_cmds[11]’) [enabled by default]
aisimage.c:34:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:34:2: warning: (near initialization for ‘aisimage_cmds[11]’) [enabled by default]
aisimage.c:35:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:35:2: warning: (near initialization for ‘aisimage_cmds[12]’) [enabled by default]
aisimage.c:35:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:35:2: warning: (near initialization for ‘aisimage_cmds[12]’) [enabled by default]
aisimage.c:35:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:35:2: warning: (near initialization for ‘aisimage_cmds[12]’) [enabled by default]
aisimage.c:36:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:36:2: warning: (near initialization for ‘aisimage_cmds[13]’) [enabled by default]
aisimage.c:36:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:36:2: warning: (near initialization for ‘aisimage_cmds[13]’) [enabled by default]
aisimage.c:36:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:36:2: warning: (near initialization for ‘aisimage_cmds[13]’) [enabled by default]
aisimage.c:37:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:37:2: warning: (near initialization for ‘aisimage_cmds[14]’) [enabled by default]
aisimage.c:37:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:37:2: warning: (near initialization for ‘aisimage_cmds[14]’) [enabled by default]
aisimage.c:37:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:37:2: warning: (near initialization for ‘aisimage_cmds[14]’) [enabled by default]
aisimage.c:38:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:38:2: warning: (near initialization for ‘aisimage_cmds[15]’) [enabled by default]
aisimage.c:38:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:38:2: warning: (near initialization for ‘aisimage_cmds[15]’) [enabled by default]
aisimage.c:38:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:38:2: warning: (near initialization for ‘aisimage_cmds[15]’) [enabled by default]
aisimage.c:39:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:39:2: warning: (near initialization for ‘aisimage_cmds[16]’) [enabled by default]
aisimage.c:39:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:39:2: warning: (near initialization for ‘aisimage_cmds[16]’) [enabled by default]
aisimage.c:39:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:39:2: warning: (near initialization for ‘aisimage_cmds[16]’) [enabled by default]
aisimage.c:40:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:40:2: warning: (near initialization for ‘aisimage_cmds[17]’) [enabled by default]
aisimage.c:40:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:40:2: warning: (near initialization for ‘aisimage_cmds[17]’) [enabled by default]
aisimage.c:40:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:40:2: warning: (near initialization for ‘aisimage_cmds[17]’) [enabled by default]
aisimage.c:41:2: warning: braces around scalar initializer [enabled by default]
aisimage.c:41:2: warning: (near initialization for ‘aisimage_cmds[18]’) [enabled by default]
aisimage.c:41:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:41:2: warning: (near initialization for ‘aisimage_cmds[18]’) [enabled by default]
aisimage.c:41:2: warning: excess elements in scalar initializer [enabled by default]
aisimage.c:41:2: warning: (near initialization for ‘aisimage_cmds[18]’) [enabled by default]
aisimage.c: In function ‘aisimage_print_header’:
aisimage.c:151:4: warning: implicit declaration of function ‘get_table_entry_name’ [-Wimplicit-function-declaration]
aisimage.c:151:4: warning: format ‘%s’ expects argument of type ‘char *’, but argument 3 has type ‘int’ [-Wformat]
aisimage.c: In function ‘aisimage_generate’:
aisimage.c:317:5: warning: implicit declaration of function ‘get_table_entry_id’ [-Wimplicit-function-declaration]
aisimage.c:335:25: error: request for member ‘sname’ in something not a structure or union
aisimage.c: In function ‘aisimage_check_image_types’:
aisimage.c:366:14: error: ‘IH_TYPE_AISIMAGE’ undeclared (first use in this function)
aisimage.c:366:14: note: each undeclared identifier is reported only once for each function it appears in
aisimage.c:370:1: warning: control reaches end of non-void function [-Wreturn-type]
make[2]: *** [aisimage.o] Error 1
make[2]: Leaving directory `/home/nyx/sunxi/u-boot/tools'
make[1]: *** [tools] Error 2
make[1]: Leaving directory `/home/nyx/sunxi/u-boot'
make: *** [sun4i] Error 2
```

**Comment by oliv3r** [comment]
Please take this to the mailinglist or IRC. U-boot seems to build fine for everybody on every board and haven't heard of any bugs in u-boot. Hence this is a request for help and not a bug in u-boot.

We gladly can talk more on IRC or on the Mailinglist but also mention your compiler, OS etc etc to get a more complete picture of where things could go wrong for you.

Please note that, atleast on Gentoo, there seems to be an issue with lensfun which contaminates your /usr/include with its own version of image.h, see http://forums.gentoo.org/viewtopic-t-935766-start-0.html for more info about this.

Basically this is a clash between u-boot-tools and lensfun-dev package. Not many users will probably hit this bug as they wont' have the lensfun-dev package (e.g. the lensfun header files).

**Comment by codyps** [comment]
I'm well aware we _could_ require users to remove a package just to get u-boot-tools to compile, but that isn't a solution, that is a workaround. If "image.h" were part of, say, libc we wouldn't even consider it. Further, this will crop up for other headers used internally by u-boot that also happen to clash with system supplied headers.

"Contaminates" is not even close to correct: u-boot-tools should be able to build irregardless of me having extra headers in /usr/include. What is really happening is that u-boot is missordering it's include path when calling gcc. The forum post you link states this explicitly and provides a solution for the missordering.

**Comment by oliv3r** [comment]
You are of course 100% right, and I apologize for my initial comment. But you are still 'barking up the wrong tree' here ;) This is really an upstream u-boot issue and should be brought there :(
